### PR TITLE
[ Milestone 4 ] Join table

### PR DESCRIPTION
**Pull request by CakeInTech**
- Created `vets` and `specialization` tables.
- Used primary key and foreign key to join the tables.
- Written queries to answer the analytic questions. 

# Queries

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -43,3 +43,34 @@ UPDATE animals SET owner_id = 2 WHERE name IN ('Gabumon', 'Pikachu');
 UPDATE animals SET owner_id = 3 WHERE name IN ('Devimon', 'Plantmon'); 
 UPDATE animals SET owner_id = 4 WHERE name IN ('Charmander', 'Squirtle', 'Blossom'); 
 UPDATE animals SET owner_id = 5 WHERE name IN ('Angemon', 'Boarmon'); 
+
+INSERT INTO vets(name, age, date_of_graduation)
+VALUES ('Vet William Tatcher', '45',  date '2000-04-23'),
+('Vet Maisy Smith', '26',  date '2019-01-17'),
+('Vet Stephanie Mendez', '64',  date '1981-05-04'),
+('Vet Jack Harkness', '38',  date '2008-06-08');
```
Comment (cynthiainga) [inline data.sql]: - [x] Please kindly correct your name values. The names are `William Tacher, Maisy Smith, Stephanie Mendez, and Jack Harkness` there is no `vet` before the names. Kindly fix that 🙃

File: queries.sql [hunk 2]
```diff
@@ -56,4 +56,21 @@ SELECT species.name, COUNT(animals.species_id) FROM animals JOIN species ON spec
...
 SELECT animals.name from animals JOIN owners ON owners.id = animals.owners_id JOIN species on species.id = animals.species_id WHERE animals.species_id = '2' AND animals.owners_id = '2';
 SELECT animals.name from animals JOIN owners ON owners.id = animals.owners_id WHERE animals.escape_attempts = '0' AND animals.owners_id = '5';
-SELECT full_name, COUNT(owners_id) FROM owners JOIN animals on owners.id = animals.owners_id GROUP BY full_name ORDER BY COUNT (owners_id) desc limit 1;
\ No newline at end of file
+SELECT full_name, COUNT(owners_id) FROM owners JOIN animals on owners.id = animals.owners_id GROUP BY full_name ORDER BY COUNT (owners_id) desc limit 1;
+
+
+----
+
+SELECT animals.name FROM animals JOIN visits ON animals.id = visits.animals_id WHERE vets_id = '1' ORDER BY visits.date_of_visit desc limit 1;
```
Comment (cynthiainga) [inline queries.sql:64]: - [x] This query should return the last animal seen by William Tatcher, and it is `Blossom` but your query return `Ditto`, `Ditto` has never been visited by William Tatcher. Please kindly write correctly your query 🙃 🙃

File: queries.sql [hunk 3]
```diff
@@ -56,4 +56,21 @@ SELECT species.name, COUNT(animals.species_id) FROM animals JOIN species ON spec
...
+SELECT COUNT(DISTINCT animals_id) FROM visits WHERE vets_id = '2';
```
Comment (cynthiainga) [inline queries.sql]: - [x] This query is supposed to return the different animals that Stephanie Mendez sees. First, on your table, the vets_id of Stephanie Mendez is `3` and not `2`. Also, you should join the animals and vets table to visits. Kindly fix that 😉 
The query can be similar to this 👇 
```SQL
 SELECT DISTINCT animals.name
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE vets.name = 'Stephanie Mendez';
```

File: queries.sql [hunk 4]
```diff
@@ -56,4 +56,21 @@ SELECT species.name, COUNT(animals.species_id) FROM animals JOIN species ON spec
...
+
+SELECT vets.name, specialization.vets_id, specialization.species_id FROM vets LEFT JOIN specialization ON specialization.vets_id = vets.id;
+SELECT animals.name , visits.date_of_visit FROM animals JOIN visits ON animals.id = visits.animals_id WHERE visits.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30' AND vets_id = '3';
```
Comment (cynthiainga) [inline queries.sql:69]: - [x] This query should return all animals that visited Stephanie Mendez between April 1st and August 30th, 2020; and is two animals `Agumon` and ` Blossom` but your query return `Agumon` and `Ditto`. `Ditto` has never been visited by Stephanie Mendez, please kindly fix that 🙃 🙃

File: queries.sql [hunk 5]
```diff
@@ -56,4 +56,21 @@ SELECT species.name, COUNT(animals.species_id) FROM animals JOIN species ON spec
...
+
+SELECT animals.name, COUNT(animals_id) FROM animals JOIN visits ON animals.id = visits.animals_id GROUP BY animals.name ORDER BY COUNT(animals_id) desc limit 1;
+SELECT animals.name, date_of_visit FROM animals JOIN visits ON animals.id = visits.animals_id WHERE vets_id = '2'ORDER BY visits.date_of_visit ASC LIMIT '1';
+
+SELECT animals.name, vets.name, visits.date_of_visit FROM animals JOIN visits ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE visits.date_of_visit = (SELECT MIN(date_of_visit) FROM visits);
```
Comment (cynthiainga) [inline queries.sql:74]: - [x] This query should return the details for the most recent visit, the most recent visit is `2021-05-04` by `Stephanie Mendez` and the animal is `Devimon`, but your query return "`Boarmon` `Vet Maisy Smith` `2019-01-24`". It should return the most recent visit and not the first visit 🙃 😉

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @MohamedCK, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] Descriptive PR ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] Good commit history ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (CakeInTech) [reply data.sql]: Thanks for the reminder, It's done. 😄

File: queries.sql (hunk 2 above)
Comment (CakeInTech) [reply queries.sql:64]: @cynthiainga  I deleted the whole data in the table animals, was part of the task on the last project but I didn't start a transaction so I had to write it all over again, and forgot to add blossom to the table. But yeah thanks for the reminder. 😃

File: queries.sql (hunk 4 above)
Comment (CakeInTech) [reply queries.sql:69]: image

Alright had to update my DB, But yeah now it returns blossom. 😄

File: queries.sql (hunk 3 above)
Comment (CakeInTech) [reply queries.sql]: image

It's done, Thanks for the reminder. 😄

File: queries.sql (hunk 5 above)
Comment (CakeInTech) [reply queries.sql:74]: Thanks for the reminder 😄

**Review by algerina — APPROVED** [review]
Hi @MohamedCK 
Good job so far! 

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- Your project looks awesome:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous TSE





 
 ## Optional suggestions
 
 - Please check the comment under the review.
 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

**Pull request merged**